src/data/zmod/basic.lean

```diff
@@ -818,32 +819,53 @@ begin
     rw [int.cast_coe_nat, int.cast_coe_nat, nat_cast_self, sub_zero, nat_cast_zmod_val], }
 end
 
+lemma val_min_abs_neg_of_ne_half {n : ℕ} {a : zmod n} (ha : 2 * a.val ≠ n) :
+  (-a).val_min_abs = -a.val_min_abs :=
+begin
+  cases n, { refl },
+  simp only [zmod.val_min_abs_def_pos, zmod.neg_val],
+  split_ifs with h h' h'' h'' h' h'' h''; try { subst h },
+  { simp },
+  { contradiction },
+  { contradiction },
+  { simp only [zero_le', not_true] at h', contradiction },
+  { exfalso,
+    by_cases hpar : even n.succ,
+    { apply ha,
+      rw [←eq_of_ge_of_not_gt h'' (not_lt_of_ge (nat.half_le_of_sub_le_half h')),
+        nat.two_mul_div_two_of_even hpar] },
+    { rw ←nat.odd_iff_not_even at hpar,
+      have := add_le_add h' h'',
+      rw [nat.sub_add_cancel (a.val_le), ←two_mul] at this,
+      have := nat.add_le_add_right this 1,
+      rw [nat.two_mul_div_two_add_one_of_odd hpar, ←not_lt] at this,
+      exact this (lt_add_one _) } },
+  { rw [neg_sub, nat.cast_sub], apply le_of_lt a.val_lt },
+  { rw nat.cast_sub (le_of_lt a.val_lt), apply sub_sub_cancel_left },
+  { exfalso, exact h'' (nat.le_half_of_half_lt_sub (not_le.mp h')) }
+end
+
+lemma val_min_abs_neg_of_eq_half {n : ℕ} {a : zmod n} (ha : 2 * a.val = n) :
```

Comment:
Just to check; `eq_half` is a deliberate naming of `2 * a.val = n` even though it doesn't actually mention `/ 2`; as opposed to being left over from a previous iteration?

Reply:
Yes, but only due to me not being able to come up with a good alternative. `...of_two_mul_eq` maybe?